server/data/data-file/src/main/java/org/apache/james/mailrepository/file/MBoxMailRepository.java

```diff
@@ -251,18 +251,9 @@ private MimeMessage convertTextToMimeMessage(String emailBody) {
 
     /**
      * Generate a hex representation of a SHA-256 checksum on the email body
-     * 
-     * @param emailBody
-     * @return A hex representation of the text
-     * @throws NoSuchAlgorithmException
      */
-    private String generateKeyValue(String emailBody) throws NoSuchAlgorithmException {
-        byte[] digArray = MessageDigest.getInstance("SHA-256").digest(emailBody.getBytes());
-        StringBuilder digest = new StringBuilder();
-        for (byte aDigArray : digArray) {
-            digest.append(Integer.toString(aDigArray, Character.MAX_RADIX).toUpperCase(Locale.US));
-        }
-        return digest.toString();
+    private String generateKeyValue(String emailBody) {
+        return Hashing.sha256().hashUnencodedChars(emailBody).toString();
```

Comment:
did you ensure the result is uppercase? can we have at least one unit test (before/after)?

Reply:
do we care ? the point is to be consistent upon the way we hash, not to respect a given format.

Reply:
that's the question. It was done before, is there any part of the code that is expecting it to be uppercase too?

Reply:
checked : it's used as an opaque value, no uppercase expected